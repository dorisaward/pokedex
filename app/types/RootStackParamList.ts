export type RootStackParamList = {
  Search: undefined;
  Details: { pokemonName: string };
  Favourites: { pokemonName: string } | undefined;
};
