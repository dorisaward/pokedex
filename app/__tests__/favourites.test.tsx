import Favourites from "@/app/favourites";
import RootStackParamList from "@/app/types/RootStackParamList";
import { RouteProp } from "@react-navigation/core";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { render, cleanup } from "@testing-library/react-native";

afterEach(cleanup);

describe("favourites", () => {
  it("renders", () => {
    // Given
    const navigation: NativeStackNavigationProp<
      RootStackParamList,
      "Favourites",
      undefined
    > = {} as NativeStackNavigationProp<
      RootStackParamList,
      "Favourites",
      undefined
    >;
    const route: RouteProp<RootStackParamList, "Favourites"> = {
      key: "Favourites",
      name: "Favourites",
      params: undefined,
      path: undefined,
    };
    const renderable = <Favourites navigation={navigation} route={route} />;

    // When
    const { toJSON } = render(renderable);

    // Then
    expect(toJSON()).toMatchSnapshot();
  });
  it("renders, given a pokemon name", () => {
    // Given
    const pokemonName = "mew";
    const navigation: NativeStackNavigationProp<
      RootStackParamList,
      "Favourites",
      undefined
    > = {} as NativeStackNavigationProp<
      RootStackParamList,
      "Favourites",
      undefined
    >;
    const route: RouteProp<RootStackParamList, "Favourites"> = {
      key: "Favourites",
      name: "Favourites",
      params: { pokemonName },
      path: undefined,
    };
    const renderable = <Favourites navigation={navigation} route={route} />;

    // When
    const { toJSON } = render(renderable);

    // Then
    expect(toJSON()).toMatchSnapshot();
  });
});
