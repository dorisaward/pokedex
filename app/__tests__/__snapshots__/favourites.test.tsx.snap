// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`favourites renders 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text />
</View>
`;

exports[`favourites renders, given a pokemon name 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    mew
  </Text>
</View>
`;
